examples/dreamcast/pvr/pvrmark_strips_direct/pvrmark_strips_direct.c

```diff
@@ -120,23 +125,30 @@ void do_frame(void) {
         vert->x = x;
         vert->y = y;
         vert->z = z;
-        vert->u = vert->v = 0.0f;
         vert->argb = col | (col << 8) | (col << 16) | 0xff000000;
-        vert->oargb = 0;
-
-        if(i == (polycnt - 1))
-            vert->flags = PVR_CMD_VERTEX_EOL;
-
         pvr_dr_commit(vert);
     }
 
+    x = (x + ((getnum(64)) - 32)) & 1023;
+    nextnum();
+    y = (y + ((getnum(64)) - 32)) % 511;
+    nextnum();
+    col = getnum(256);
+    nextnum();
+    vert = pvr_dr_target(dr_state);
+    vert->flags = PVR_CMD_VERTEX_EOL;
+    vert->x = x;
+    vert->y = y;
+    vert->z = z;
+    vert->argb = col | (col << 8) | (col << 16) | 0xff000000;
+    pvr_dr_commit(vert);
```

Comment:
Instead of duplicating this, would it make sense to just be a function that the loop calls and is called for cleanup? Could be inlined to not have the overhead. 

At the very least, it seems like that first vert could be part of the loop by just moving the seed adjusting part to the end of the loop rather than the start so that it uses the right initial values. That'd have the side effect of leaving the seeds incremented for that final vert too.

Reply:
Yeah, you're right. Cleaned.